aulas/04.md

```diff
@@ -154,17 +171,36 @@ table_registry = registry()
 class User:
     __tablename__ = 'users'
 
-    id: Mapped[int] = mapped_column(init=False, primary_key=True)
-    username: Mapped[str] = mapped_column(unique=True)
+    id: Mapped[int] = mapped_column(init=False, primary_key=True)#(1)!
+    username: Mapped[str] = mapped_column(unique=True)#(2)!
     password: Mapped[str]
     email: Mapped[str] = mapped_column(unique=True)
-    created_at: Mapped[datetime] = mapped_column(
+    created_at: Mapped[datetime] = mapped_column(#(3)!
         init=False, server_default=func.now()
     )
 ```
 
-Uma menção especial ao campo `init`. Ele diz que quando a classe é instanciada, o SQLAlchemy irá atribuir um valor automaticamente para o atributo. No caso das chaves primárias, elas serão incrementadas automaticamente. O campo `created_at` executa a função atrelada ao parâmetro `server_default`, no caso atribuindo `func.now()` ao campo.
+1. `#!python init=False` diz que, quando o objeto for instanciado, esse parâmetro não deve ser passado. `#!python primary_key=True` diz que o campo `id` é a chave primária dessa tabela.
+2. `unique=True` diz que esse campo não deve se repetir na tabela. Por exemplo, se tivermos um username "dunossauro", não podemos ter outro com o mesmo valor.
+3. `server_default=func.now()` diz que, quando a classe for instanciada, o resultado de `func.now()` será o valor atribuído a esse atributo. No caso, a data e hora em que ele foi instanciado.
+
+Desta forma, unimos tanto o uso que queremos ter no python, quanto a configuração esperada da tabela no banco de dados. Os parâmetros de mapeamento dizem:
+
+- `primary_key`: diz que, o campo será a chave primária da tabela
+- `unique`: diz que o campo só pode ter um valor único em toda a tabela. Não podemos ter um `username` repetido no banco, por exemplo.
```

Comment:
Nessa linha falta uma vírgula depois do que no inicio

Reply:
Duno eu consigo fazer a mudança? Negocio tao besta mas nao achei onde modificar

Reply:
Olá @azmovi tem uma doc do github que é bem legal e ensina isso de fazer sugestões de alteração no PR, segue o link: https://docs.github.com/pt/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request

Sobre a sua observação da vírgula eu penso que o correto retirar a vírgula na linha do primary_key. Para mim fica mais fluída a leitura sem a vírgula nesses casos. A vírgula meio que isola o "diz que"

Reply:
Valeu vou dar uma olhada, então eu confie na vírgula do duno na linha de cima, por mim ficaria sem virgula tbm